backend/apps/medical_catalog/services/hospital_item_price_service.py

```
@@ -1,9 +1,61 @@
 
+def _is_unique_constraint_violation(error: IntegrityError) -> bool:
+    """
+    Determine if an IntegrityError is a unique constraint violation.
+    
+    Checks database-specific error codes and constraint names to positively
+    identify unique constraint violations, avoiding false positives from
+    NOT NULL, foreign key, or other constraint violations.
+    """
+    # Check for PostgreSQL unique violation (pgcode 23505)
+    if hasattr(error, 'orig') and hasattr(error.orig, 'pgcode'):
+        if error.orig.pgcode == '23505':  # unique_violation
```

Ruff (0.14.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)